BrainPortal/app/views/tasks/show.html.erb

```diff
@@ -57,6 +57,26 @@
       <% confirm += "\n\nIMPORTANT NOTE: This will also remove its archived work directory." if @task.workdir_archive %>
       <%= link_to 'Remove Task',  { :action => 'operation', :operation => 'delete', 'tasklist[]' => @task.id }, :confirm => confirm, :class  => "button menu_button", :method => :post %>
     <% end %>
+    <% if @task.status =~ /Failed*|Completed/ %>
+      <%= link_to "Remove Work Directory", url_for(:action  => :operation, :operation => 'zap_wd', 'tasklist[]' => @task.id),
+                                                  :method  => :post,
+                                                  :datatype => :script,
+                                                  :class => "button",
+                                                  :confirm =>
+                  "Are you sure you want to remove the task's work directory?"
+      %>
+    <% end %>
+
+    <!-- ARCHIVING -->
+    <% if @task.status =~ /Completed/ && @task.full_cluster_workdir.blank? == false && @task.archived_status.nil? %>
+      <%= link_to "Archive on Cluster", url_for(:action  => :operation, :operation => 'archive', 'tasklist[]' => @task.id), :method  => :post, :datatype => :script, :class => "button" %>
+      <% if @task.results_data_provider_id %>
```

Comment:
Andrew, this hasn't been addressed. I was saying you cannot have a "Archive As File" button that defaults to the results_data_provider_id for the file's destination. Either we have no button for this action, or you provide a selector for the data provider.